src/cached_object_store/ghost.rs

```diff
@@ -0,0 +1,196 @@
+use lru::LruCache;
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::num::NonZeroUsize;
+
+/// The weighter is used to calculate the weight of the cache entry.
+pub trait Weighter<K, V>: Fn(&K, &V) -> usize + Send + Sync + 'static {}
+impl<K, V, T> Weighter<K, V> for T where for<'a> T: Fn(&K, &V) -> usize + Send + Sync + 'static {}
+
+/// A simple wrapper of GhostQueue based on LRU eviction
+pub struct GhostQueue<K: Hash + Eq, V> {
+    queue: LruCache<K, V>,
+    capacity: NonZeroUsize,
+    weighter: Box<dyn Weighter<K, V>>,
+    usage: usize,
```

Comment:
is this usage a bytes size?

Reply:
Yes.

Reply:
To be precise, usage is related to weighter calculation. If the default weighter is used, it is the number of entries.